R/map_epiflows.R

```diff
@@ -1,153 +1,125 @@
+#'          "LUX", "MEX", "MMR", "NIC", "NLD", "PAN", "PER", "PRT", "PRY",
+#'          "SLV", "SWE", "TTO", "USA", "VEN"),
+#'   lon = c(-63.616672, 4.469936, -77.39628, -88.49765, -63.588653,
+#'           -51.92528, -59.543198, -106.346771, 8.227512, -71.542969,
+#'           104.195397, -74.297333, -83.753428, -77.781167, 10.451526,
+#'           -70.162651, -78.183406, -3.74922, 25.7481511, 2.213749, -3.435973,
+#'           -90.230759, 114.109497, -86.241905, 12.56738, -77.297508, 138.252924,
+#'           6.129583, -102.552784, 95.955974, -85.207229, 5.291266,
+#'           -80.782127, -75.015152, -8.224454, -58.443832, -88.89653, 18.643501,
+#'           -61.222503, -95.712891, -66.58973),
+#'   lat = c(-38.416097, 50.503887, 25.03428, 17.189877, -16.290154,
+#'           -14.235004, 13.193887, 56.130366, 46.818188, -35.675147,
+#'           35.86166, 4.570868, 9.748917, 21.521757, 51.165691, 18.735693,
+#'           -1.831239, 40.463667, 61.92411, 46.227638, 55.378051, 15.783471,
+#'           22.396428, 15.199999, 41.87194, 18.109581, 36.204824, 49.815273,
+#'           23.634501, 21.916221, 12.865416, 52.132633, 8.537981, -9.189967,
+#'           39.399872, -23.442503, 13.794185, 60.128161, 10.691803, 37.09024,
+#'           6.42375)
+#'   )
+#' flows   <- Mex_travel_2009[[1]]
+#' outflow <- setNames(flows[["MEX"]], rownames(flows))
+#' inflow  <- unlist(flows["MEX", , drop = TRUE])
+#' ef      <- epiflows(inflow, outflow, focus = "MEX", locations = Mex_travel_2009[[2]])
+#' ef      <- add_coordinates(ef, loc[-1])
+#' map_epiflows(ef, center = "MEX")
+map_epiflows <- function(x, title = "", center = NULL, sort = TRUE, ...) {
 
-#' @export
-map_epiflows <- function(x,
-                          type = "foo",
-                          origin,
-                          title = sprintf("Flows from %s", origin_name),
-                          loc_column = "country",
-                          lon_lat_columns = c("lon", "lat"),
-                          ...) {
-  # Add coordinates if needed
-  location_cols <- names(x$locationsdata)
-  if (!loc_column %in% location_cols) {
-    stop("`%s` not found in x$locationsdata")
+  # First thing to do is to calculate the great circle arc for the flows with
+  # the make_lines internal function.
+  the_flows <- get_flows(ef)
+  if (sort) {
+    the_flows <- the_flows[order(the_flows$n), ]
   }
+  the_coords  <- get_coords(ef)
+  
+  # SpatialLinesDataFrame class construction ------------------------
+  sldf <- make_SpatialLinesDataFrame(the_flows, the_coords)
 
-  if (!all(lon_lat_columns %in% location_cols)) {
-    ## Tries to find the coordinates with add_coordinates(), should only do this if type == "map"
-    if(type=="map"){
-     x %<>%
-      add_coordinates(
-        loc_column = loc_column,
-        lon_lat_columns = lon_lat_columns
-      )
+  # Leaflet plot construction ---------------------------------------
+  pal    <- leaflet::colorQuantile(palette = "Greys",
```

Comment:
I don't mind the grey, but people typically like pretty colors; maybe default to something more colorful, and pass it through as an argument to `map_epiflows(pal = "whatever")`; depending on how many palettes are available, maybe list them in the argument and use `match.arg`? Not super hard set on this, happy to chat

Reply:
I've changed the color palette to a yellow-orange-brown palette and also added line thickness which is `ln(n + 1)`